File: api/api-javascript/api-utils/src/main/resources/utils/v3/digest.js
/*
 * Copyright (c) 2010-2020 SAP and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * Contributors:
 *   SAP - initial API and implementation
 */
var java = require('core/v3/java');
var streams = require('io/v3/streams');

exports.md5 = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'md5', [JSON.stringify(bytes)]);
	return output;
};

exports.md5Hex = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'md5Hex', [JSON.stringify(bytes)]);
	return output;
};

exports.sha1 = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'sha1', [JSON.stringify(bytes)]);
	return output;
};

exports.sha256 = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'sha256', [JSON.stringify(bytes)]);
	return output;
};

exports.sha384 = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'sha384', [JSON.stringify(bytes)]);
	return output;
};

exports.sha512 = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'sha512', [JSON.stringify(bytes)]);
	return output;
};

exports.sha1Hex = function(input) {
	var bytes = input;
	if (typeof bytes === 'string') {
		var baos = streams.createByteArrayOutputStream();
		baos.writeText(bytes);
		bytes = baos.getBytes();
	}
	var output = java.call('org.eclipse.dirigible.api.v3.utils.DigestFacade', 'sha1Hex', [JSON.stringify(bytes)]);
	return output;
};

File: components/api/api-modules-python/src/main/resources/META-INF/dirigible/python-modules/dirigible/utils/digest.py
import hashlib

def md5(input):
    if isinstance(input, str):
        input = input.encode('utf-8')
    md5_hash = hashlib.md5(input)
    return md5_hash.digest()

def md5_hex(input):
    if isinstance(input, str):
        input = input.encode('utf-8')
    md5_hash = hashlib.md5(input)
    return md5_hash.hexdigest()

def sha1(input):
    if isinstance(input, str):
        input = input.encode('utf-8')
    sha1_hash = hashlib.sha1(input)
    return sha1_hash.digest()

def sha1_hex(input):
    if isinstance(input, str):
        input = input.encode('utf-8')
    sha1_hash = hashlib.sha1(input)
    return sha1_hash.hexdigest()

def sha256(input):
    if isinstance(input, str):
        input = input.encode('utf-8')
    sha256_hash = hashlib.sha256(input)
    return sha256_hash.digest()

def sha384(input):
    if isinstance(input, str):
        input = input.encode('utf-8')
    sha384_hash = hashlib.sha384(input)
    return sha384_hash.digest()

def sha512(input):
    if isinstance input, str:
        input = input.encode('utf-8')
    sha512_hash = hashlib.sha512(input)
    return sha512_hash.digest()
